### PR TITLE
buildbot: 0.9.7 -> 0.9.9.post2

### PR DESCRIPTION
**Pull request by nand0p**
- adds grid-view plugin
- module fixup
- tested on nixos

### CODE REVIEW

File: pkgs/development/python-modules/pympler/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  version = "0.4.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0mhyxqlkha98y8mi5zqcjg23r30mgdjdzs05lghbmqfdyvzjh1a3";
+  };
+
+  doCheck = !stdenv.isDarwin;
+
+  # Remove test asizeof.flatsize(), broken and can be missed as
```
Comment (0xABAB) [inline pkgs/development/python-modules/pympler/default.nix]: Not a sentence and please contact upstream for this.

File: pkgs/development/tools/build-managers/buildbot/default.nix [hunk 2]
```diff
@@ -83,7 +83,7 @@ let
 
     meta = with stdenv.lib; {
       homepage = http://buildbot.net/;
-      description = "Continuous integration system that automates the build/test cycle";
+      description = "Buildbot is an open-source continuous integration framework for automating software build, test, and release processes.";
```
Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/default.nix]: I like this.

File: pkgs/development/tools/build-managers/buildbot/plugins.nix [hunk 3]
```diff
@@ -1,85 +1,82 @@
...
+      inherit pname version format;
+      sha256 = "0nsx4b1mrli89rikb1r91rbnyr8j8afpzplcnj0d1ivydbw6ldks";
     };
 
-    propagatedBuildInputs = with pythonPackages; [ setuptools ];
-
     meta = with stdenv.lib; {
       homepage = http://buildbot.net/;
-      description = "Buildbot Packaging Helper";
+      description = "Buildbot UI";
```
Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: Do you mean a web interface?

File: pkgs/development/tools/build-managers/buildbot/plugins.nix [hunk 4]
```diff
@@ -1,85 +1,82 @@
...
     src = pythonPackages.fetchPypi {
       inherit pname version format;
-      sha256 = "1wf2spnilm0dkyw95vf57lca453sbly4r6ak3lxa8bpwhxb6lkdn";
+      sha256 = "1hs27zw0qwgnhrz9xsmhx4ahcz885pqxxxq6h51cxhdigwvsinc1";
     };
 
     meta = with stdenv.lib; {
       homepage = http://buildbot.net/;
-      description = "Buildbot UI";
+      description = "Buildbot Console View Plugin";
```
Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: Provides a command line interface for Buildbot

File: pkgs/development/tools/build-managers/buildbot/plugins.nix [hunk 5]
```diff
@@ -1,85 +1,82 @@
...
+      inherit pname version format;
+      sha256 = "1jifcvvvfal38avizrxj10bmmi19s84yzva41q5v30sdnriim6ib";
     };
 
-    propagatedBuildInputs = with pythonPackages; [ buildbot-pkg ];
-
     meta = with stdenv.lib; {
       homepage = http://buildbot.net/;
-      description = "Buildbot Console View Plugin";
+      description = "Buildbot Waterfall View Plugin";
```
Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: Can we get a long description too, since https://pypi.python.org/pypi/buildbot-waterfall-view also doesn't bother to give one?

File: nixos/modules/services/continuous-integration/buildbot/master.nix [hunk 6]
```diff
@@ -225,11 +225,7 @@ in {
...
-
-        # NOTE: call twistd directly with stdout logging for systemd
-        #ExecStart = "${cfg.package}/bin/buildbot start --nodaemon ${cfg.buildbotDir}";
-        ExecStart = "${pkgs.python27Packages.twisted}/bin/twistd -n -l - -y ${cfg.buildbotDir}/buildbot.tac";
+        ExecStart = "${cfg.package}/bin/buildbot start --nodaemon ${cfg.buildbotDir}";
```
Comment (0xABAB) [inline nixos/modules/services/continuous-integration/buildbot/master.nix]: ${cfg.buildbotDir} doesn't work when it contains spaces.

File: pkgs/development/python-modules/pympler/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  patchPhase = ''
```
Comment (FRidh) [inline pkgs/development/python-modules/pympler/default.nix]: postPatch

File: pkgs/development/python-modules/pympler/default.nix [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  name = "${pname}-${version}";
+  pname = "Pympler";
+  version = "0.4.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0mhyxqlkha98y8mi5zqcjg23r30mgdjdzs05lghbmqfdyvzjh1a3";
+  };
+
+  doCheck = !stdenv.isDarwin;
```
Comment (FRidh) [inline pkgs/development/python-modules/pympler/default.nix]: Why not on Darwin? Leave a comment.

File: pkgs/development/python-modules/pympler/default.nix [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  # Remove test asizeof.flatsize(), broken and can be missed as
+  patchPhase = ''
+    substituteInPlace ./test/asizeof/test_asizeof.py --replace "n, e = test_flatsize" "#n, e = test_flatsize"
+    substituteInPlace ./test/asizeof/test_asizeof.py --replace "self.assert_(n," "#self.assert_(n,"
+    substituteInPlace ./test/asizeof/test_asizeof.py --replace "self.assert_(not e" "#self.assert_(not e"
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://pythonhosted.org/Pympler/;
+    description = "A development tool to measure, monitor and analyze the memory behavior of Python objects.";
```
Comment (FRidh) [inline pkgs/development/python-modules/pympler/default.nix]: no dot at the end

File: pkgs/development/tools/build-managers/buildbot/default.nix (hunk 2 above)
Comment (FRidh) [inline pkgs/development/tools/build-managers/buildbot/default.nix]: no dot at the end

File: pkgs/development/tools/build-managers/buildbot/plugins.nix [hunk 10]
```diff
@@ -1,85 +1,82 @@
 { stdenv, pythonPackages }:
 
-let
-  buildbot-pkg = pythonPackages.buildPythonPackage rec {
+# NOTE: buildbot plugins available as wheels
```
Comment (FRidh) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: They only distribute wheels? Bleh...

**Review by FRidh — CHANGES_REQUESTED** [review]
Thanks for the update. Some minor fixes are needed and then I think its ready to go in.

File: pkgs/development/tools/build-managers/buildbot/plugins.nix (hunk 10 above)
Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: No, they just don't follow their own documentation: https://github.com/buildbot/buildbot/issues/3409.

Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: For the main release there is now a tar.gz. I didn't check for the plugins; that might be different.

File: pkgs/development/tools/build-managers/buildbot/plugins.nix [hunk 11]
```diff
@@ -4,11 +4,11 @@ let
   buildbot-pkg = pythonPackages.buildPythonPackage rec {
     name = "${pname}-${version}";
     pname = "buildbot-pkg";
-    version = "0.9.7";
+    version = "0.9.9.post1";
 
     src = pythonPackages.fetchPypi {
       inherit pname version;
-      sha256 = "0p351r10y42gwgxb2qg7xlsbhmnzdmqp6h4922l0yfii3pzmrdzv";
+      sha256 = "0000000000000000000000000000000000000000000000000000";
```
Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: That's unlikely to be the correct hash.

File: pkgs/development/tools/build-managers/buildbot/plugins.nix [hunk 12]
```diff
@@ -25,14 +25,14 @@ in {
   www = pythonPackages.buildPythonPackage rec {
     name = "${pname}-${version}";
     pname = "buildbot_www";
-    version = "0.9.7";
+    version = "0.9.9.post1";
 
     # NOTE: wheel is used due to buildbot circular dependency
```
Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: Is anyone supposed to understand what you mean by that comment?

File: pkgs/development/tools/build-managers/buildbot/plugins.nix (hunk 11 above)
Comment (nand0p) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: waiting on the buildbot maintainers @tardyp to publish the source tarballs, so as to update this hash

File: pkgs/development/tools/build-managers/buildbot/plugins.nix (hunk 12 above)
Comment (nand0p) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: `.post1` is not a comment, it is the actual release version

File: nixos/modules/services/continuous-integration/buildbot/master.nix (hunk 6 above)
Comment (nand0p) [inline nixos/modules/services/continuous-integration/buildbot/master.nix]: yes, i think that is the correct expected behavior on linux/unix systems.

File: pkgs/development/tools/build-managers/buildbot/plugins.nix (hunk 12 above)
Comment (0xABAB) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: @nand0p I was talking about the only comment visible, which I suppose is not part of this PR, but still. 
` # NOTE: wheel is used due to buildbot circular dependency`

File: pkgs/development/tools/build-managers/buildbot/plugins.nix [hunk 13]
```diff
@@ -82,4 +82,25 @@ in {
...
   };
+
+  grid-view = pythonPackages.buildPythonPackage rec {
+    name = "${pname}-${version}";
+    pname = "buildbot-grid-view";
+    version = "0.9.9.post1";
+
+    src = pythonPackages.fetchPypi {
+      inherit pname version;
+      sha256 = "0000000000000000000000000000000000000000000000000000";
```
Comment (Mic92) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: this hash is not correct.

File: pkgs/development/tools/build-managers/buildbot/plugins.nix (hunk 12 above)
Comment (Mic92) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: Please change your tone in pull request reviews. This is not the linux ml. The semantics of `format` is documented in `buildPythonPackage`.

File: nixos/modules/services/continuous-integration/buildbot/master.nix (hunk 6 above)
Comment (0xABAB) [inline nixos/modules/services/continuous-integration/buildbot/master.nix]: How can you expect something not to work? It clearly can work by properly escaping the argument.

Comment (edolstra) [inline nixos/modules/services/continuous-integration/buildbot/master.nix]: `buildbotDir` containing whitespace didn't work before this PR, so it's not really the job of this PR to fix it.

File: pkgs/development/tools/build-managers/buildbot/plugins.nix (hunk 11 above)
Comment (FRidh) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: If the lack of sdist is preventing this from going in, then let's just use wheels or tarballs from GitHub.

**Comment by tardyp** [comment]
Guys, I've published 0.9.9.post2 for you yesterday. 
You should have tarballs and wheels properly uploaded, on top of a couple of bugfixes.
https://pypi.python.org/pypi/buildbot/0.9.9.post2

File: nixos/modules/services/continuous-integration/buildbot/master.nix (hunk 6 above)
Comment (0xABAB) [inline nixos/modules/services/continuous-integration/buildbot/master.nix]: @edolstra OK, perhaps there has been a misunderstanding, but I expect that every replacement line committed does not contain any potential flaw. So, while it might be true that BuildBot currently does not support spaces, at some point that limitation might be gone. In that case a version with properly escaped arguments would just work. 

I don't think we should encourage bad practices. 

If, on the other hand you mean that it's OK to open PRs for things that just make things slightly better as opposed to fixing things completely, then that's a policy which seems questionable. 

The way I see it is that if an author modifies an existing line, they are reintroducing the same mistake, i.e. creating a bug. 

I have the impression that you see this differently.

Comment (nand0p) [inline nixos/modules/services/continuous-integration/buildbot/master.nix]: `you mean that it's OK to open PRs for things that just make things slightly better as opposed to fixing things completely` yes. we have been maintaining the buildbot package iteratively.

**Comment by nand0p** [comment]
@FRidh ready_for_review

File: nixos/modules/services/continuous-integration/buildbot/master.nix (hunk 6 above)
Comment (edolstra) [inline nixos/modules/services/continuous-integration/buildbot/master.nix]: @0xABAB That's right, the PR should not introduce regressions. It does not have to reach a state of perfection.

**Review by FRidh — APPROVED** [review]
